js/renovation/button.tsx

```diff
@@ -46,80 +46,52 @@ const getAriaLabel = (text, icon) => {
     return label ? { label } : {};
 };
 
-export const viewModelFunction = (model: Button):ButtonViewModel => {
-    const props = model.props;
-    return {
-        ...props,
-        aria: getAriaLabel(model.props.text, model.props.icon),
-        contentRef: model.contentRef,
-        cssClasses: getCssClasses(props),
-        elementAttr: { ...props.elementAttr, role: 'button' },
-        iconSource: (props.icon || props.type === 'back') ? (props.icon || 'back') : '',
-        onActive: model.onActive,
-        onInactive: model.onInactive,
-        onWidgetClick: model.onWidgetClick,
-        onWidgetKeyPress: model.onWidgetKeyPress,
-        submitInputRef: model.submitInputRef,
-    };
-};
-
-declare type ButtonViewModel = {
-    contentRef: any;
-    cssClasses: string;
-    iconSource: string;
-    onActive: (e: Event) => any;
-    onInactive: (e: Event) => any;
-    onWidgetClick: (e: Event) => any;
-    onWidgetKeyPress: (e: Event, options:any) => void;
-    submitInputRef: any;
-} & ButtonInput;
-
-export const viewFunction = (viewModel: ButtonViewModel) => {
-    const renderText = !viewModel.contentRender && viewModel.text;
-    const isIconLeft = viewModel.iconPosition === 'left';
-    const leftIcon = !viewModel.contentRender && isIconLeft;
-    const rightIcon = !viewModel.contentRender && !isIconLeft;
-    const icon = !viewModel.contentRender && viewModel.iconSource
-        && <Icon source={viewModel.iconSource} position={viewModel.iconPosition}/>;
+export const viewFunction = (viewModel: Button) => {
+    const renderText = !viewModel.props.contentRender && viewModel.props.text;
+    const isIconLeft = viewModel.props.iconPosition === 'left';
+    const leftIcon = !viewModel.props.contentRender && isIconLeft;
```

Comment:
Why don't you use object destruction?

```js
({ props: { text, ... }, ... })
```

or

```js
const { text, ... } = viewModel.props;
```

Reply:
I did not write this code. Refactoring goes out of the context of this PR

Reply:
Okay